formats/json/jvmMain/src/kotlinx/serialization/json/internal/JvmJsonStreams.kt

```diff
@@ -0,0 +1,266 @@
+package kotlinx.serialization.json.internal
+
+import java.io.InputStream
+import java.io.OutputStream
+import java.nio.charset.Charset
+
+internal class JsonToJavaStreamWriter(private val stream: OutputStream) : JsonWriter {
+    private val buffer = ByteArrayPool.take()
+    private var array = CharArrayPool.take()
+    private var index: Int = 0
```

Comment:
I'd suggest renaming to 'byteIndex` or smth to clearly indicate which array this index for.